vision_cnn:
    ros__parameters:
        # LDC LUT definition
        lut_file_path: /opt/robotics_sdk/ros2/drivers/zed_capture/config/SN5867575_HD_LUT_right.bin

        # location of network
        # Known Issue: ONR-OD-8080-yolov3 has initial loading time ~20 seconds
        dl_model_path: /opt/model_zoo/ONR-OD-8080-yolov3-lite-regNetX-1.6gf-bgr-coco-512x512
        # dl_model_path: /opt/model_zoo/TFL-OD-2020-ssdLite-mobDet-DSP-coco-320x320

        ## Models below are all detection models from edge AI model zoo. Not all models yet tested.
        ## Please download the models first using /opt/edge_ai_apps/download_models.sh
        # dl_model_path: /opt/model_zoo/ONR-OD-8000-ssd1200-resNet34-mlperf-1200x1200
        # dl_model_path: /opt/model_zoo/ONR-OD-8020-ssd-lite-mobv2-coco-512x512
        # dl_model_path: /opt/model_zoo/ONR-OD-8030-ssd-lite-mobv2-fpn-coco-512x512
        # dl_model_path: /opt/model_zoo/ONR-OD-8040-ssd-lite-regNetX-200mf-fpn-bgr-coco-320x320
        # dl_model_path: /opt/model_zoo/ONR-OD-8050-ssd-lite-regNetX-800mf-fpn-bgr-coco-512x512
        # dl_model_path: /opt/model_zoo/ONR-OD-8060-ssd-lite-regNetX-1.6gf-bifpn168x4-bgr-coco-768x768
        # dl_model_path: /opt/model_zoo/ONR-OD-8080-yolov3-lite-regNetX-1.6gf-bgr-coco-512x512
        # dl_model_path: /opt/model_zoo/ONR-OD-8090-retina-lite-regNetX-800mf-fpn-bgr-coco-512x512
        # dl_model_path: /opt/model_zoo/TFL-OD-2000-ssd-mobV1-coco-mlperf-300x300
        # dl_model_path: /opt/model_zoo/TFL-OD-2010-ssd-mobV2-coco-mlperf-300x300
        # dl_model_path: /opt/model_zoo/TFL-OD-2020-ssdLite-mobDet-DSP-coco-320x320
        # dl_model_path: /opt/model_zoo/TFL-OD-2030-ssdLite-mobDet-EdgeTPU-coco-320x320
        # dl_model_path: /opt/model_zoo/TFL-OD-2060-ssdlite-mobV2-coco-300x300
        # dl_model_path: /opt/model_zoo/TFL-OD-2070-ssd-mobV1-fpn-coco-tpu-8-640x640
        # dl_model_path: /opt/model_zoo/TFL-OD-2080-ssd-mobV2-fpnlite-coco-tpu-8-320x320
        # dl_model_path: /opt/model_zoo/TFL-OD-2090-ssd-mobV2-fpnlite-coco-tpu-8-640x640
        # dl_model_path: /opt/model_zoo/TFL-OD-2110-efficient-det-ti-lite0-512x512
        # dl_model_path: /opt/model_zoo/TFL-OD-2130-ssd-mobV2-coco-tpu-8-320x320
        # dl_model_path: /opt/model_zoo/TVM-OD-5020-yolov3-mobv1-gluon-mxnet-416x416

        # input data width in pixels
        width:                  1280

        # input data height in lines
        height:                 720

        # input image format
        #  0 - VX_DF_IMAGE_UYVY
        #  1 - VX_DF_IMAGE_NV12
        image_format:           0

        # Maximum number of Semantic Segmentation classes
        num_classes:            20

        # interactive mode
        is_interactive:         0

        # To enable showing output on display when display_option = 1
        display_option:         1

        # pipeline depth
        pipeline_depth:         4

        # Flag to indicate if the graph should be exported
        #  0 - disable
        #  1 - enable
        exportGraph:            0

        # Flag to indicate if the performance data should be exported
        # to a file
        #  0 - disable
        #  1 - enable
        exportPerfStats:        0

        # Real-time logging enable.
        #  0 - disable
        #  1 - enable
        rtLogEnable:            0

        # log level for the app
        # 0 - DEBUG: Used to show general debug messages
        # 1 - INFO: Used to show run-time processing debug
        # 2 - WARN: Used to warning developers of possible issues
        # 3 - ERROR: Used for most errors
        log_level:              2
